test/distributed/org/apache/cassandra/distributed/test/accord/AccordMetricsTest.java

```diff
@@ -0,0 +1,258 @@
+        assertCoordinatorMetrics(1, "ro", 0, 0, 0, 0, 1);
+        assertReplicaMetrics(0, "ro", 1, 1, 0);
+        assertReplicaMetrics(1, "ro", 1, 1, 0);
+
+        assertZeroMetrics("rw");
+    }
+
+    @Test
+    public void testTimeoutMetrics()
+    {
+        IMessageFilters.Filter preAcceptFilter = SHARED_CLUSTER.filters().outbound().verbs(Verb.ACCORD_PRE_ACCEPT_REQ.id).from(1).to(2).drop();
+        preAcceptFilter.on();
+
+        countingMetrics0 = getMetrics();
+        try
+        {
+            SHARED_CLUSTER.coordinator(1).executeWithResult(readCql(), ConsistencyLevel.ALL, 0, 0);
+        }
+        catch (RuntimeException t)
+        {
+            assertThat(t.getCause().getClass().getName()).isEqualTo(ReadTimeoutException.class.getName());
+        }
+
+        assertCoordinatorMetrics(0, "ro", 0, 0, 0, 1, 0);
+        assertCoordinatorMetrics(1, "ro", 0, 0, 0, 0, 0);
+        assertReplicaMetrics(0, "ro", 0, 0, 0);
+        assertReplicaMetrics(1, "ro", 0, 0, 0);
+
+        assertZeroMetrics("rw");
+
+        countingMetrics0 = getMetrics();
+        try
+        {
+            SHARED_CLUSTER.coordinator(1).executeWithResult(writeCql(), ConsistencyLevel.ALL, 0, 0, 0, 0);
+        }
+        catch (RuntimeException t)
+        {
+            assertThat(t.getCause().getClass().getName()).isEqualTo(WriteTimeoutException.class.getName());
+        }
+
+        assertCoordinatorMetrics(0, "rw", 0, 0, 0, 1, 0);
+        assertCoordinatorMetrics(1, "rw", 0, 0, 0, 0, 0);
+        assertReplicaMetrics(0, "rw", 0, 0, 0);
+        assertReplicaMetrics(1, "rw", 0, 0, 0);
+
+        assertZeroMetrics("ro");
+    }
+
+    private void assertZeroMetrics(String scope)
+    {
+        for (int i = 0; i < SHARED_CLUSTER.size(); i++)
+        {
+            assertCoordinatorMetrics(i, scope, 0, 0, 0, 0, 0);
+            assertReplicaMetrics(i, scope, 0, 0, 0);
+        }
+    }
+
+    private void assertCoordinatorMetrics(int node, String scope, long fastPathTrx, long slowPathTrx, long preemptions, long timeouts, long recoveries)
+    {
+        DefaultNameFactory nameFactory = new DefaultNameFactory("accord-coordinator", scope);
+        Map<String, Long> metrics = diff(countingMetrics0).get(node);
+        logger.info("Metrics for node {} / {}: {}", node, scope, metrics);
+        Function<String, Long> metric = n -> metrics.get(nameFactory.createMetricName(n).getMetricName());
+        assertThat(metric.apply("FastPathTrx")).isEqualTo(fastPathTrx);
+        assertThat(metric.apply("SlowPathTrx")).isEqualTo(slowPathTrx);
+        assertThat(metric.apply("PreemptedTrx")).isEqualTo(preemptions);
+        assertThat(metric.apply("TimedoutTrx")).isEqualTo(timeouts);
+        assertThat(metric.apply("RecoveryDelay")).isEqualTo(recoveries);
+        assertThat(metric.apply("RecoveryTime")).isEqualTo(recoveries);
+        assertThat(metric.apply("Dependencies")).isEqualTo(fastPathTrx + slowPathTrx);
```

Comment:
nit: Should we have perhaps a smoke test to ensure the `FastPathToTotalX` bits are published?